ci/do_ci.sh

```diff
@@ -303,6 +303,7 @@ function bazel_envoy_api_go_build() {
 
 CI_TARGET=$1
 shift
+CI_PARAMS=("$@")
```

Comment:
this would ignore the first argument - not sure if that is what we want you would need to do something like ...

```console
./ci/do_ci.sh IGNOREME arg1 arg2 ...
```

Reply:
actually - ignore me - CI_TARGET is `refresh_compdb` in this case so above comment is wrong

Reply:
I am sure. First arg is the target name and I was going to ignore it.

Reply:
i guess im a little uncomfortable more generally with making do_ci.sh into an arg accepting command

atm its more for generating standard args to pass to tooling - which is why it doesnt have `${@}` passthrough already

Reply:
I am open to that. What's your suggestion I I want to do similar things? Create a new script under the `ci/` or `tools/`

Reply:
well i would say that do_ci.sh should probs work with standard args - as used in ci - and for other uses of the tool its better to call directly

but then brings us back to original reason for using it - to get the build_setup - so not sure

Reply:
one option is to do something like what do_ci.sh does for bazel args - ie define them as env var and then parse those into an array

Reply:
It is okay for me for this PR only.

But I have another point. I hope we can provide a CLI tools that could help developers to develop, build, test, check format check, check clang tidy, etc., as supplements to poor C++ tool chains. Parameters could be used to control different functions. The developers needn't to find different helper scripts at every where of the repo.

Now, the do_ci.sh is something that similar to my expected role. Although it's not designed to do this.

Reply:
yeah - its not designed to do this

the role of this file is to hold the canonical ci targets - as called in actual ci

this role has become even more important as the workflow files are not editable in a pr - so the workflows hand over to these targets

Reply:
tbh - the role you are describing is more bazel's role